scripts/generate-install-yaml.sh

```diff
@@ -20,10 +20,33 @@ set -o pipefail
 
 INSTALL_YAML="${INSTALL_YAML:-hack/install-latest.yaml}"
 IMAGE_NAME="${IMAGE_NAME:-quay.io/kubernetes-multicluster/federation-v2:latest}"
+FEDERATION_NAMESPACE="${NS:-""}"
 
 kubebuilder create config --controller-image "${IMAGE_NAME}" \
             --name federation --output "${INSTALL_YAML}"
 
 # Increase memory request and limit to avoid OOM issues.
 sed -i 's/memory: 20Mi/memory: 64Mi/' "${INSTALL_YAML}"
 sed -i 's/memory: 30Mi/memory: 128Mi/' "${INSTALL_YAML}"
+
+# Remove namespace resource from the top of the file
+sed -i -e '/---/,$!d' "${INSTALL_YAML}"
+
+# Remove namespace fields
+sed -i -e '/^  namespace: federation-system$/d' "${INSTALL_YAML}"
+
+if [ "${FEDERATION_NAMESPACE}" ]; then
+  # Update name for ClusterRole
+  sed -i -e 's/name: federation-role$/name: federation-role-'${FEDERATION_NAMESPACE}'/' "${INSTALL_YAML}"
```

Comment:
Why is it desirable to name the cluster role and role binding with the namespace?  A cluster-scoped federation can only be deployed once to a cluster, so finding a name clash would be a good sign of cleanup required rather than a problem.

Reply:
Deleted.